internal/provider/waypoint/data_source_waypoint_add_on_definition.go

```diff
@@ -0,0 +1,225 @@
+	TerraformNoCodeModule   *tfcNoCodeModule `tfsdk:"terraform_no_code_module"`
+}
+
+func NewAddOnDefinitionDataSource() datasource.DataSource {
+	return &DataSourceAddOnDefinition{}
+}
+
+func (d *DataSourceAddOnDefinition) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_waypoint_add_on_definition"
+}
+
+func (d *DataSourceAddOnDefinition) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "The Waypoint Add-on Definition data source retrieves information on a given Add-on Definition.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:    true,
+				Optional:    true,
+				Description: "The ID of the Add-on Definition.",
+			},
+			"name": schema.StringAttribute{
+				Description: "The name of the Add-on Definition.",
+				Computed:    true,
+				Optional:    true,
+			},
+			"organization_id": schema.StringAttribute{
+				Description: "The ID of the HCP organization where the Waypoint Add-on Definition is located.",
+				Computed:    true,
+			},
+			"project_id": schema.StringAttribute{
+				Description: "The ID of the HCP project where the Waypoint Add-on Definition is located.",
+				Optional:    true,
+				Computed:    true,
+			},
+			"summary": schema.StringAttribute{
+				Description: "A short summary of the Add-on Definition.",
+				Computed:    true,
+			},
+			"description": schema.StringAttribute{
+				Description: "A longer description of the Add-on Definition.",
+				Computed:    true,
+			},
+			"readme_markdown_template": schema.StringAttribute{
+				Computed:    true,
+				Description: "Instructions for using the definition (markdown format supported)",
+			},
+			"labels": schema.ListAttribute{
+				Computed:    true,
+				Description: "List of labels attached to this Add-on Definition.",
+				ElementType: types.StringType,
+			},
+			"terraform_cloud_workspace_details": &schema.SingleNestedAttribute{
+				Computed:    true,
+				Description: "Terraform Cloud Workspace details",
+				Attributes: map[string]schema.Attribute{
+					"name": &schema.StringAttribute{
+						Computed:    true,
+						Description: "Name of the Terraform Cloud Workspace",
+					},
+					"terraform_project_id": &schema.StringAttribute{
+						Computed:    true,
+						Description: "Tetraform Cloud Project ID",
+					},
+				},
+			},
+			"terraform_no_code_module": &schema.SingleNestedAttribute{
+				Computed:    true,
+				Description: "Terraform Cloud No Code Module details",
+				Attributes: map[string]schema.Attribute{
+					"source": &schema.StringAttribute{
+						Computed:    true,
+						Description: "No Code Module Source",
+					},
+					"version": &schema.StringAttribute{
+						Computed:    true,
+						Description: "No Code Module Version",
```

Comment:
I lowercased this and added a dash, as this is what I've observed in the TFC documentation.

[In this tutorial, you will review a no-code module](https://developer.hashicorp.com/terraform/tutorials/cloud/no-code-provisioning)...

```suggestion
				Description: "Terraform Cloud no-code module details",
				Attributes: map[string]schema.Attribute{
					"source": &schema.StringAttribute{
						Computed:    true,
						Description: "Terraform Cloud no-code module source",
					},
					"version": &schema.StringAttribute{
						Computed:    true,
						Description: "Terraform Cloud no-code module version",
```